app/containers/ITunesContainer/tests/index.test.js

```diff
@@ -84,4 +93,65 @@ describe('<ITunesContainer /> container tests', () => {
     props.dispatchClearITunesSongs();
     expect(dispatchITunesSongsSpy).toHaveBeenCalledWith(actions.dispatchClearITunesSongs);
   });
+
+  it('should render default error message when search goes wrong', () => {
+    const defaultError = translate('something_went_wrong');
+    const { getByTestId } = renderProvider(<ITunesContainer iTunesError={defaultError} />);
+    expect(getByTestId('error-message')).toBeInTheDocument();
+    expect(getByTestId('error-message').textContent).toBe(defaultError);
+  });
+
+  it('should render the data when loading becomes false', () => {
+    const iTunesData = { totalCount: 1, results: [{ artistName: 'Jaymes Young' }] };
+    const { getByTestId } = renderProvider(<ITunesContainer iTunesData={iTunesData} dispatchITunesSongs={submitSpy} />);
+    expect(getByTestId('for')).toBeInTheDocument();
+  });
+
+  it('should render exact number of iTuneCards as per totalCount in result', () => {
+    const totalCount = 2;
+    const iTunesData = {
+      totalCount,
+      results: [
+        {
+          artistName: 'Mark Ronson',
+          trackName: 'Uptown Funk (feat. Bruno Mars)',
+          primaryGenreName: 'Pop'
+        },
+        {
+          artistName: 'Dynamix Music',
+          trackName: 'Uptown Funk',
+          primaryGenreName: 'Fitness & Workout'
+        }
+      ]
+    };
+    const { getAllByTestId } = renderProvider(
+      <ITunesContainer iTunesData={iTunesData} dispatchITunesSongs={submitSpy} />
+    );
+    expect(getAllByTestId('itunes-card').length).toBe(totalCount);
+  });
+
+  it('should render Skeleton Comp when "loading" is true', async () => {
+    const searchQuery = 'Infinity';
+    const { getByTestId, baseElement } = renderProvider(
+      <ITunesContainer dispatchITunesSongs={submitSpy} searchQuery={searchQuery} />
+    );
+    fireEvent.change(getByTestId('search-bar'), { target: { value: searchQuery } });
+    await timeout(500);
+    expect(baseElement.getElementsByClassName('ant-skeleton').length).toBe(1);
+  });
+
+  it('should test play/pause functionality', async () => {
+    const iTunesData = { totalCount: 2, results: [{ artistName: 'Jaymes' }, { artistName: 'Young' }] };
+    const { getAllByTestId } = renderProvider(
+      <ITunesContainer iTunesData={iTunesData} dispatchITunesSongs={submitSpy} />
+    );
+    const players = getAllByTestId('itunes-card');
+    const buttons = getAllByTestId('play-button');
+
+    buttons.forEach((item) => {
+      fireEvent.click(item);
+    });
+
+    expect(players[0].querySelector('audio')).toHaveProperty('paused', true);
```

Comment:
here we aren't testing that the other ones are still playing. We should be testing that right? 

so if we click play on one card, only that card is playing. If we click play on another card, the first card pauses and the second card plays. 

Test for outcome

Reply:
Okay.added test to check other card  is  playing